apps/transformers/lib/parse_utils.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Transformers.ParseUtils do
+  def operandsToNumbers(operands, payload) when is_list(operands) do
+    parseValues(operands, payload)
+  end
+
+  def operandsToNumbers(operands, payload) when is_binary(operands) do
+    String.split(operands, [" ", ","], trim: true)
+    |> parseValues(payload)
+  end
+
+  def operandsToNumbers(_, _) do
+    {:error, "Operands must be a list of values or a string representation of a list of values"}
+  end
+
+  def parseValues(operands, payload) do
+    result =
+      Enum.reduce_while(operands, [], fn operand, acc ->
+        case parseValue(operand, payload) do
+          {:ok, number} -> {:cont, [number | acc]}
+          {:error, reason} -> {:halt, {:error, reason}}
+        end
+      end)
+
+    case result do
+      {:error, reason} -> {:error, reason}
+      _ -> {:ok, Enum.reverse(result)}
+    end
+  end
+
+  def parseValue(value, _) when is_number(value) do
+    {:ok, value}
+  end
+
+  def parseValue(value, payload) do
```

Comment:
I don't love how this function operates at the moment. Ideally we wouldn't need to call all three parse functions, but would instead just return the one that is needed. This doesn't seem possible without enforcing field types on the transformation input parameters so we know which function to call.

It would be possible to just return the first function that succeeds, but I couldn't find a way of doing that without a triple nested conditional, so I went for clarity over performance here.

I'd welcome any alternative ideas people have.

Reply:
What about using [Decimal](https://hexdocs.pm/decimal/Decimal.html) which seems to be a BigDecimal equivalent from other languages? Looks like we already have that wired in for the division transformation.